iceoryx_posh/source/roudi/memory/posix_shm_memory_provider.cpp

```diff
@@ -33,7 +33,7 @@ void sigbusHandler(int32_t) noexcept
         "memory. Please make sure that enough memory is available. For this, consider also the memory which is "
         "required for the [/iceoryx_mgmt] segment. Please refer to share/doc/iceoryx/FAQ.md in your release delivery.";
     size_t len = strlen(msg);
-    size_t count [[gnu::unused]] = write(STDERR_FILENO, msg, len);
+    ssize_t count [[gnu::unused]] = write(STDERR_FILENO, msg, len);
```

Comment:
would `auto` also work?
This reminds me that we need a DISCARD_RESULT macro for cases like this

Reply:
how do you like this
```
#define DISCARD_RESULT_VARIABLE_GENERATOR(name, suffix) name ## _ ## suffix
#define DISCARD_RESULT_VARIABLE(name, suffix) DISCARD_RESULT_VARIABLE_GENERATOR(name, suffix)
#define DISCARD_RESULT(expr) auto DISCARD_RESULT_VARIABLE(unusedOnLine, __LINE__) [[gnu::unused]] = expr
```

Reply:
Looks quite promising, maybe ican put that into iceoryx_utils/cxx/helplets.hpp?